Author: aszlig <aszlig@nix.build>
Date:   Thu Apr 5 07:56:12 2018 +0200

    Improve and fix nightwatch config
    
    This gets rid of webpack, because we already have a running production
    instance in our VM test, so no need pack the client once again.
    
    Furthermore, we don't start Selenium, because this is done by the VM
    test as well.
    
    And the last thing is that we add nightwatch-html-reporter to
    package.json, so that it is picked up by node2nix.
    
    Signed-off-by: aszlig <aszlig@nix.build>
    Filename: nightwatch.patch

diff --git a/package.json b/package.json
index d3eb01f8e..8a8c532cc 100644
--- a/package.json
+++ b/package.json
@@ -165,6 +165,7 @@
     "mocha-simple-html-reporter": "^1.1.0",
     "monk": "^6.0.6",
     "nightwatch": "^0.9.21",
+    "nightwatch-html-reporter": "^2.0.5",
     "puppeteer": "^1.5.0",
     "require-again": "^2.0.0",
     "selenium-server": "^3.12.0",
diff --git a/test/client/e2e/nightwatch.conf.js b/test/client/e2e/nightwatch.conf.js
index 251dd8bbc..6b0360e9c 100644
--- a/test/client/e2e/nightwatch.conf.js
+++ b/test/client/e2e/nightwatch.conf.js
@@ -1,7 +1,6 @@
 /* eslint-disable camelcase */
 
 require('babel-register');
-const config = require('../../../webpack/config');
 const chromeDriverPath = require('chromedriver').path;
 const seleniumServerPath = require('selenium-server').path;
 
@@ -12,7 +11,7 @@ module.exports = {
   custom_assertions_path: ['test/client/e2e/custom-assertions'],
 
   selenium: {
-    start_process: true,
+    start_process: false,
     server_path: seleniumServerPath,
     host: '127.0.0.1',
     port: 4444,
@@ -27,7 +26,7 @@ module.exports = {
       selenium_host: 'localhost',
       silent: true,
       globals: {
-        devServerURL: `http://localhost:${process.env.PORT || config.dev.port}`, // eslint-disable-line no-process-env
+        devServerURL: 'http://localhost',
       },
     },
 
diff --git a/test/client/e2e/runner.js b/test/client/e2e/runner.js
index 4d40c020d..b449c9097 100644
--- a/test/client/e2e/runner.js
+++ b/test/client/e2e/runner.js
@@ -1,8 +1,4 @@
-// 1. start the dev server using production config
-process.env.NODE_ENV = 'testing'; // eslint-disable-line no-process-env
-const server = require('../../../webpack/dev-server.js');
-
-// 2. run the nightwatch test suite against it
+// run the nightwatch test suite against it
 // to run in additional browsers:
 //    1. add an entry in test/e2e/nightwatch.conf.json under "test_settings"
 //    2. add it to the --env flag below
@@ -21,11 +17,9 @@ const spawn = require('cross-spawn');
 const runner = spawn('nightwatch', opts, { stdio: 'inherit' });
 
 runner.on('exit', function (code) {
-  server.close();
   process.exit(code);
 });
 
 runner.on('error', function (err) {
-  server.close();
   throw err;
 });
